src/Activity.js

```diff
@@ -1,50 +1,95 @@
 class Activity {
-  constructor(activityData) {
-    this.activityData = activityData;
+  constructor(activityData, user) {
+
+    //TODO: change to userId
+    this.user = user;
+    this.activityData = this.getUserActivityData(activityData);
+  }
+
+  getUserActivityData(activityData) {
+    return activityData.filter((data) => {
+      return this.user.id === data.userID;
+    });
+  }
+
+  getActivityOnDate(date) {
+    return this.activityData.find((data) => {
+      return date === data.date;
+    })
   }
 
-  getMilesFromStepsByDate(id, date, userRepo) {
-    const userStepsByDate = this.activityData.find((data) => id === data.userID && date === data.date);
-    return parseFloat(((userStepsByDate.numSteps * userRepo.strideLength) / 5280).toFixed(1));
+  getMilesFromStepsByDate(date) {
+    let activityLog = this.getActivityOnDate(date);
+    let stepsPerMile = (5280 / this.user.strideLength);
+    let milesWalked = (activityLog.numSteps / stepsPerMile);
+    return (Math.round(milesWalked * 10) / 10);
   }
 
-  getActiveMinutesByDate(id, date) {
-    const userActivityByDate = this.activityData.find((data) => id === data.userID && date === data.date);
-    return userActivityByDate.minutesActive;
+  getActiveMinutesByDate(date) {
+    let activityLog = this.getActivityOnDate(date);
+    return activityLog.minutesActive;
   }
 
-  calculateActiveAverageForWeek(id, date, userRepo) {
-    return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => acc += elem.minutesActive, 0) / 7).toFixed(1));
+  calculateActiveAverageForWeek(date) {
+    let week = this.findGivenWeek(date);
+    let totalTime = week.reduce((totalMinutes, element) => {
+      totalMinutes += element.minutesActive;
+      return totalMinutes;
+    }, 0);
+    return (Math.round((totalTime / 7) * 10) / 10);
```

Comment:
I'm having trouble wrapping my head around multiplying by 10 and then dividing by 10. Gonna look at this one more so I understand it better!

Reply:
Useful for rounding!

Reply:
Got a great explanation from @Shawntru Thanks! It is so that the number is rounded to the nearest 10th! Sweet!